### PR TITLE
Github orgs are now requested async

### LINKED ISSUE #48: Async fetch of organization when linking to Github
In the view to link to Github a new Organization, the available ones in Github are listed. This is done in the same request and the user has to wait for this.

The code looks like:

``` ruby
def new
    authorize organization, :update?
    organization
    # TODO: this should be done over ajax so the user does not notice the delay
    @github_orgs = github_service.get_organizations(per_page: 400)
    @github_orgs.select! { |o| !existing_organizations.include?(o.login) }
  end
```

It should be done using ajax and get all the available ones, instead of 400

### PR DESCRIPTION
**Pull request by mdesanti**
## Summary

Now github organizations are retrieved asynchronously

Closes #48

### CODE REVIEW

**Comment by mdesanti** [comment]
@epintos in the CI build you can see the error with `rollbar.rb` i was telling you about

File: .gitignore [hunk 1]
```diff
@@ -38,3 +38,5 @@ test
 .byebug_history
 
 .env.local
+.idea/*
```
Comment (epintos) [inline .gitignore]: What is this?

Comment (mdesanti) [inline .gitignore]: I was trying Intellij as an alternative IDE

File: app/assets/javascripts/exponential_backoff.js.coffee [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    () ->
+      configObject.retryIn *= 1.25
+      organizations.exponentialBackoff(url, callback, configObject)
+  , configObject.retryIn
+  )
+
+setDefaults = (configObject) ->
+  configObject = configObject || {}
+  configObject.retries = configObject.retries || 0
+  configObject.retryIn = configObject.retryIn || 2000
```
Comment (epintos) [inline app/assets/javascripts/exponential_backoff.js.coffee]: consts

File: app/assets/javascripts/exponential_backoff.js.coffee [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    datatype : "application/json"
+    statusCode: {
+      202: (data, textStatus, jqXHR) ->
+        retry(url, callbacks, configObject)
+      200: (data, textStatus, jqXHR) ->
+        callbacks.success(data)
+    }
+    error: (data, textStatus, jqXHR) ->
+      console.log('error')
+  })
```
Comment (epintos) [inline app/assets/javascripts/exponential_backoff.js.coffee]: I think you are missing an enter at the end of the files

File: app/assets/javascripts/exponential_backoff.js.coffee [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    url: url
+    datatype : "application/json"
+    statusCode: {
+      202: (data, textStatus, jqXHR) ->
+        retry(url, callbacks, configObject)
+      200: (data, textStatus, jqXHR) ->
+        callbacks.success(data)
+    }
+    error: (data, textStatus, jqXHR) ->
+      console.log('error')
```
Comment (epintos) [inline app/assets/javascripts/exponential_backoff.js.coffee]: Hmm it would be nice to set the error in the .alert div that we have

Comment (epintos) [inline app/assets/javascripts/exponential_backoff.js.coffee]: I think you have the error and you didn't use it

File: app/assets/javascripts/exponential_backoff.js.coffee (hunk 2 above)
Comment (mdesanti) [inline app/assets/javascripts/exponential_backoff.js.coffee]: This code was reviewed by sebas balay in this pr or another project https://github.com/Wolox/projects-manager/pull/5

**Comment by epintos** [comment]
👍

**Comment by epintos** [comment]
👍

**Pull request merged**